agent/agent.go

```diff
@@ -973,14 +975,16 @@ func (a *Agent) finishTask(ctx context.Context, tc *taskContext, status string,
 		if err := a.runPostOrTeardownTaskCommands(ctx, tc); err != nil {
 			tc.logger.Task().Info("Post task completed - FAILURE. Overall task status changed to FAILED.")
 			setEndTaskFailureDetails(tc, detail, evergreen.TaskFailed, "", "")
+			detail.ErrorBlockType = string(tc.getCurrentBlock())
```

Comment:
I suppose I'm fine with this approach, but what this approach needs to accomplish is it needs to be able to indicate whether post commands failed even when the task succeeded (i.e., even if `post_error_fails_task` = false). The casing [here](https://gist.github.com/hadjri/8b8b99be1521cd43af9c65a3f82c8a4f#file-agent-go-L859) will need to check to update the failing block type regardless of whether `post.canFailTask`

Reply:
Ah I see, I did not realize that's the point. I think I agree with your earlier comment more now to just simplify it and to update that casting you linked to set it as well!